NPE on Block Break, related to async lighting

SpongeVanilla version: 7.1.0-BETA-99.

On the server there are no plugins.

Breaking blocks randomly throws NPE. I was just building something, then looked to the console and saw this exception.

Block that caused this error will not be saved. Restarting server will load that block normally like it was never broken.

```
[00:16:23] [Server thread/FATAL]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_171]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_171]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) ~[h.class:?]
	at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:293) ~[SpongeImplHooks.class:1.12.2-7.1.0-BETA-99]
	at net.minecraft.server.MinecraftServer.redirect$onRun$zjf000(SourceFile:4123) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(SourceFile:1581) ~[MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(SourceFile:349) ~[nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(SourceFile:560) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(SourceFile:464) ~[MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.NullPointerException
	at net.minecraft.server.management.PlayerInteractionManager.handler$onBlockClicked$bac000(SourceFile:1056) ~[or.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.func_180784_a(SourceFile) ~[or.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147345_a(SourceFile:638) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayerDigging.func_148833_a(SourceFile:40) ~[lp.class:?]
	at net.minecraft.network.play.client.CPacketPlayerDigging.func_148833_a(SourceFile:10) ~[lp.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:92) ~[PacketUtil.class:1.12.2-7.1.0-BETA-99]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zla000(SourceFile:539) ~[hv$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_171]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_171]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 7 more
```

Remapped:
```
[00:16:23] [Server thread/FATAL]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_171]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_171]
	at net.minecraft.util.Util.runTask(SourceFile:47) ~[h.class:?]
	at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:293) ~[SpongeImplHooks.class:1.12.2-7.1.0-BETA-99]
	at net.minecraft.server.MinecraftServer.redirect$onRun$zjf000(SourceFile:4123) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(SourceFile:1581) ~[MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(SourceFile:349) ~[nz.class:?]
	at net.minecraft.server.MinecraftServer.tick(SourceFile:560) ~[MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(SourceFile:464) ~[MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.NullPointerException
	at net.minecraft.server.management.PlayerInteractionManager.handler$onBlockClicked$bac000(SourceFile:1056) ~[or.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.onBlockClicked(SourceFile) ~[or.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processPlayerDigging(SourceFile:638) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayerDigging.processPacket(SourceFile:40) ~[lp.class:?]
	at net.minecraft.network.play.client.CPacketPlayerDigging.processPacket(SourceFile:10) ~[lp.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:92) ~[PacketUtil.class:1.12.2-7.1.0-BETA-99]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zla000(SourceFile:539) ~[hv$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_171]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_171]
	at net.minecraft.util.Util.runTask(SourceFile:46) ~[h.class:?]
	... 7 more
```

--- a/src/main/java/org/spongepowered/server/mixin/core/server/management/MixinPlayerInteractionManager.java
+++ b/src/main/java/org/spongepowered/server/mixin/core/server/management/MixinPlayerInteractionManager.java
@@ -30,6 +30,7 @@ import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumHand;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
 import org.spongepowered.api.block.BlockSnapshot;
 import org.spongepowered.api.world.Location;
 import org.spongepowered.api.world.World;
@@ -52,8 +53,9 @@ public abstract class MixinPlayerInteractionManager {
     public void onBlockClicked(BlockPos pos, EnumFacing side, CallbackInfo ci) {
         final BlockSnapshot blockSnapshot = new Location<>((World) this.player.world, VecHelper.toVector3d(pos)).createSnapshot();
         final RayTraceResult result = SpongeImplHooks.rayTraceEyes(this.player, SpongeImplHooks.getBlockReachDistance(this.player));
+        final Vec3d hitVec = result == null ? null : result.hitVec;
 
-        if (SpongeCommonEventFactory.callInteractBlockEventPrimary(this.player, blockSnapshot, EnumHand.MAIN_HAND, side, VecHelper.toVector3d(result.hitVec)).isCancelled()) {
+        if (SpongeCommonEventFactory.callInteractBlockEventPrimary(this.player, blockSnapshot, EnumHand.MAIN_HAND, side, VecHelper.toVector3d(hitVec)).isCancelled()) {
             ((IMixinEntityPlayerMP) this.player).sendBlockChange(pos, this.player.world.getBlockState(pos));
             // Copied from Forge PlayerInteractionManager
             player.world.notifyBlockUpdate(pos, player.world.getBlockState(pos), player.world.getBlockState(pos), 3);

